### PR TITLE
393 add translate by and rotate by to mobile base

### PR DESCRIPTION
**Pull request by glannuzel**
Add new functions `reachy.mobile_base.translate_by(x, y)` and `reachy.mobile_base.rotate_by(x, y)`.

No test has been added as the mobile base movements are not simulated so far.

:warning: WARNING : Not tested on the robot so far

### CODE REVIEW

File: src/reachy2_sdk/parts/mobile_base.py [hunk 1]
```diff
@@ -173,6 +173,23 @@ def set_speed(self, x_vel: float, y_vel: float, rot_vel: float) -> None:
...
+        x = odometry["x"] + x
+        y = odometry["y"] + y
+        self.goto(x, y, 0, timeout=timeout)
+
+    def rotate_by(self, theta: float, timeout: Optional[float] = None) -> None:
+        """Send a target rotation relative to the current rotation of the mobile base.
+
+        theta defines the rotation wanted in the mobile base in cartesian space."""
+        odometry = self.odometry
+        theta = odometry["theta"] + theta
```
Comment (ClaireHzl) [inline src/reachy2_sdk/parts/mobile_base.py:190]: We should add the position, or the robot will translate to get back to x=0, y = 0 if it already has been moved from its origin.
 <code> def rotate_by(self, theta: float, timeout: Optional[float] = None) -> None:
        odometry = self.odometry
        x = odometry["x"]
        y = odometry["y"]
        theta = odometry["theta"] + theta
        self.goto(x, y, theta, timeout=timeout) </code>

**Comment by ClaireHzl** [comment]
translate_by works perfectly on the real robot !
But rotate_by doesn't work well, because reachy.mobile_base.odometry['theta'] is always = 0

**Comment by glannuzel** [comment]
Waiting for PR https://github.com/pollen-robotics/reachy2_sdk_server/pull/176 to be merged on SDK server

**Comment by glannuzel** [comment]
https://github.com/pollen-robotics/reachy2_sdk_server/pull/176 has been merged!

**Comment by ClaireHzl** [comment]
Works fine on the robot but maybe the tolerance is too high on the orientation ? 
image

For the translate_by, it's always in the initial frame of the robot. So if you did a rotate_by(90) then a translate_by(x = 0.2, y =0), the robot will translate to the right and not front.

File: src/reachy2_sdk/parts/mobile_base.py [hunk 2]
```diff
@@ -173,6 +173,26 @@ def set_speed(self, x_vel: float, y_vel: float, rot_vel: float) -> None:
         )
         self._mobility_stub.SendDirection(req)
 
+    def translate_by(self, x: float, y: float, timeout: Optional[float] = None) -> None:
+        """Send a target position relative to the current position of the mobile base.
+
+        (x, y) define the translation wanted in the mobile base in cartesian space."""
+        odometry = self.odometry
```
Comment (ClaireHzl) [inline src/reachy2_sdk/parts/mobile_base.py:180]: This maybe will work better 
<code> 
def translate_by(self, x: float, y: float, timeout: Optional[float] = None) -> None:
    odometry = self.odometry
    x_current = odometry["x"]
    y_current = odometry["y"]
    theta = odometry["theta"]
    theta_rad = deg2rad(theta)
    x_goal = x_current + (x * cos(theta_rad) - y * sin(theta_rad))
    y_goal = y_current + (x * sin(theta_rad) + y * cos(theta_rad))
    self.goto(x_goal, y_goal, theta, timeout=timeout)
</code>

**Review by ClaireHzl — APPROVED** [review]
Works perfectly !

**Pull request merged**